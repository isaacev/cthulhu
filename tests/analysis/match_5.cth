type Thing =
  | Coord { x: Int, y: Int }
  | None;

#[entry]
fn main() -> () {
  let c = Thing::Cord { x: 2, y: 4 };
  match c {
    Thing::Coord { x: x, x: y } => ::Io::println("hello"),
    _ => ::Io::println("empty"),
  };
}
