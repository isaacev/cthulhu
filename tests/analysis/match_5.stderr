
  REDUNDANT NAMED FIELDS  ----------------------------------------------------

  There are more than one fields named `x`. The first usage is here:

    : match_5
  9 |     Thing::Coord { x: x, x: y } => ::Io::println("hello"),
    |                    ^

  The second usage is here:

    : match_5
  9 |     Thing::Coord { x: x, x: y } => ::Io::println("hello"),
    |                          ^

